src/library-authoring/LibraryBlock/LibraryBlock.tsx

```diff
@@ -0,0 +1,94 @@
+/* eslint-disable react/require-default-props */
+import React, { useEffect, useRef, useState } from 'react';
+import { getConfig } from '@edx/frontend-platform';
+
+interface LibraryBlockProps {
+  onBlockNotification?: (event: { eventType: string; [key: string]: any }) => void;
+  usageKey: string;
+}
+/**
+ * React component that displays an XBlock in a sandboxed IFrame.
+ *
+ * The IFrame is resized responsively so that it fits the content height.
+ *
+ * We use an IFrame so that the XBlock code, including user-authored HTML,
+ * cannot access things like the user's cookies, nor can it make GET/POST
+ * requests as the user. However, it is allowed to call any XBlock handlers.
+ */
+const LibraryBlock = ({ onBlockNotification, usageKey }: LibraryBlockProps) => {
+  const iframeRef = useRef<HTMLIFrameElement>(null);
+  const [iFrameHeight, setIFrameHeight] = useState(600);
+  const lmsBaseUrl = getConfig().LMS_BASE_URL;
+
+  /**
+   * Handle any messages we receive from the XBlock Runtime code in the IFrame.
+   * See wrap.ts to see the code that sends these messages.
+   */
+  /* istanbul ignore next */
+  const receivedWindowMessage = async (event) => {
+    if (!iframeRef.current || event.source !== iframeRef.current.contentWindow) {
+      return; // This is some other random message.
+    }
+
+    const { method, replyKey, ...args } = event.data;
+    // const frame = iframeRef.current.contentWindow;
+    // const sendReply = async (data) => {
+    //   frame?.postMessage({ ...data, replyKey }, '*');
+    // };
```

Comment:
I think we can delete this commented code?

Reply:
Fixed here: 0ed8e025efc74ea8ebf0ddd8f4e2916e34f60fdc